src/daemon/screenlock.vala

```diff
@@ -0,0 +1,312 @@
+/*
+ * This file is part of budgie-desktop
+ *
+ * Copyright Budgie Desktop Developers
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+namespace Budgie {
+    const string DBUS_SCREENLOCK = "org.buddiesofbudgie.BudgieScreenlock";
+	const string DBUS_SCREENLOCK_PATH = "/org/buddiesofbudgie/Screenlock";
+
+	[DBus (name="org.buddiesofbudgie.BudgieScreenlock")]
+    public class Screenlock {
+        static Screenlock? instance;
+
+        // connections to various schemas used in screenlocking
+        private GLib.Settings power;
+        private GLib.Settings session;
+        private GLib.Settings screensaver;
+        private GLib.Settings lockdown;
+
+        // public functions should check this variable first and exit gracefully where needed
+        bool all_apps = true;
+
+        // remember the locker to be used
+        private string locker = "";
+
+        /** Our upower client */
+        private Up.Client client;
+        private GLib.GenericArray<Up.Device> devices;
+        private bool battery_mode;
+
+        private string calculate_lockcommand() {
+            string output = "";
+
+            if (locker == "swaylock") {
+                // grab the background picture and strip the file prefix
+                string current_wallpaper = screensaver.get_string("picture-uri").replace("file://","");
+
+                // swaylock interprets : as a delimiter for the output name
+                // recommendation by swaylock is to replace : with ::
+                current_wallpaper = current_wallpaper.replace(":", "::");
+
+                File file = File.new_for_path(current_wallpaper);
+                if (current_wallpaper == "" || !file.query_exists()) {
+                    output = "-c 000000";
+                }
+                else {
+                    output = "-i " + current_wallpaper;
+                }
+
+                output = "swaylock -Fklf " + output;
```

Comment:
Better practice would be to use long-form options so it's obvious to the reader of the code what this is supposed to do, even if it makes the string longer. `-Fklf` is incomprehensible without checking `swaylock --help` - `swaylock --show-failed-attempts --show-keyboard-layout --indicator-caps-lock --daemonize` is completely self-explanatory. I would like to see long-form options used here and elsewhere in this function.